apps/gitness/src/pages/pull-request/hooks/usePRChecksDecision.ts

```diff
@@ -0,0 +1,174 @@
+import { useEffect, useMemo, useState } from 'react'
+import { TypesPullReq, TypesRepository, useChecksPullReqQuery } from '@harnessio/code-service-client'
+
+import { ExecutionState } from '../../../types'
+import { determineStatusMessage, generateStatusSummary } from '../utils'
+
+interface Check {
+  check: {
+    status: ExecutionState
+  }
+}
+
+interface Data {
+  checks: Check[]
+}
+export function usePRChecksDecision({
+  repoMetadata,
+  pullReqMetadata
+}: {
+  repoMetadata: TypesRepository | undefined
+  pullReqMetadata: TypesPullReq | undefined
+}) {
+  const { data, error, refetch } = useChecksPullReqQuery({
+    repo_ref: `${repoMetadata?.path}/+`,
+    pullreq_number: pullReqMetadata?.number ?? 0,
+    queryParams: { debounce: 500 }
+  })
+  const [count, setCount] = useState(DEFAULT_COUNTS)
+  const [color, setColor] = useState<string>('GREEN_500')
+  const [background, setBackground] = useState<string>('GREEN_50')
+  const [message, setMessage] = useState('')
+  const [complete, setComplete] = useState(true)
+  const [summaryText, setSummaryText] = useState('')
+  const [checkInfo, setCheckInfo] = useState<{
+    title: string
+    content: string
+    color: string
+    status: string
+  }>({ title: '', content: '', color: '', status: '' })
+
+  const status = useMemo(() => {
+    let _status: ExecutionState | undefined
+    const _count = { ...DEFAULT_COUNTS }
+    // @ts-expect-error remove "@ts-expect-error" once CodeServiceClient Response for useChecksPullReqQuery is fixed
+    const total = data?.checks?.length
+    // @ts-expect-error remove "@ts-expect-error" once CodeServiceClient Response for useChecksPullReqQuery is fixed
+    const { message: summaryMessage } = generateStatusSummary(data?.checks)
+    setSummaryText(summaryMessage)
+    // @ts-expect-error remove "@ts-expect-error" once CodeServiceClient Response for useChecksPullReqQuery is fixed
+    const checkInfoData = determineStatusMessage(data?.checks)
+    if (checkInfoData) {
+      setCheckInfo(checkInfoData)
+    }
+    if (total) {
+      // @ts-expect-error remove "@ts-expect-error" once CodeServiceClient Response for useChecksPullReqQuery is fixed
+      for (const check of (data as Data).checks) {
+        switch (check.check.status) {
+          case ExecutionState.ERROR:
+          case ExecutionState.FAILURE:
+          case ExecutionState.RUNNING:
+          case ExecutionState.PENDING:
+          case ExecutionState.SUCCESS:
+            _count[check.check.status]++
+            setCount({ ..._count })
+            break
+          default:
+            console.error('Unrecognized PR check status', check)
+            break
+        }
+      }
+
+      if (_count.error) {
+        _status = ExecutionState.ERROR
+        setColor('text-destructive')
+        setBackground('text-destructive')
+        setMessage(`${_count.error}/${total} ${_count.error === 1 ? 'check' : 'checks'} errored out.`)
```

Comment:
Can we move this to a "pluralize" function instead? I see this `${_count.error === 1 ? 'check' : 'checks'}1` bit being repeated multiple times.

Reply:
done